camel-core/src/main/java/org/apache/camel/builder/ExpressionBuilder.java

```diff
@@ -1863,9 +1867,30 @@ public String toString() {
         };
     }
 
+    public static Expression dateExpression(final String command) {
+        return dateExpression(command, TimeZone.getDefault().getID(), "");
+    }
+
     public static Expression dateExpression(final String command, final String pattern) {
+        return dateExpression(command, TimeZone.getDefault().getID(), pattern);
+    }
+
+    public static Expression dateExpression(final String commandWithOffsets, final String timezone, final String pattern) {
         return new ExpressionAdapter() {
             public Object evaluate(Exchange exchange) {
+                // Capture optional time offsets
+                String command = commandWithOffsets.split("[+-]", 2)[0].trim();
+                List<Long> offsets = new ArrayList<>();
+                Matcher offsetMatcher = Pattern.compile("([+-])([^+-]+)").matcher(commandWithOffsets);
+                while (offsetMatcher.find()) {
+                    try {
+                        long value = exchange.getContext().getTypeConverter().mandatoryConvertTo(long.class, exchange, offsetMatcher.group(2).trim());
+                        offsets.add(offsetMatcher.group(1).equals("+") ? value : -value);
+                    } catch (NoTypeConversionAvailableException e) {
+                        e.printStackTrace();
```

Comment:
Is it sufficient to replace the dump with `throw ObjectHelper.wrapCamelExecutionException(exchange, e);`?